src/torchcodec/decoders/_core/CMakeLists.txt

```diff
@@ -16,7 +16,13 @@ function(make_torchcodec_library library_name ffmpeg_target)
         VideoDecoder.cpp
         VideoDecoderOps.h
         VideoDecoderOps.cpp
+        DeviceInterface.h
     )
+    if(ENABLE_CUDA)
+        list(APPEND sources CudaDevice.cpp)
+    else()
+        list(APPEND sources CPUOnlyDevice.cpp)
```

Comment:
Super Nit: just curious why one is called "CudaDevice" and the other is "CPUOnlyDevice"? That seems to suggest that "CudaDevice" isn't "Cuda Only" itself?

Reply:
Yes CudaDevice will support both CPU and CUDA. For CPU it will just do what it normally does. For CUDA it will do something special.

CPUOnly will error if it sees any other device.